apps/fem/poisson3.sim

```diff
@@ -0,0 +1,188 @@
+%
+%  Strong form (poisson problem)
+%
+% - \nabla^2 u = f(x,y) in \Omega (domain)
+%            u = 0      on \Gamma (boundary)
+%
+%  solved using linear functions on a tetrahedral mesh
+%
+
+element Node
+    x    : vector[3](float); % position
+    edge : bool;             % true if part of the boundary
+    u    : float;            % solution
+    i    : int;              % index of node
```

Comment:
I see the problem, but we also need to retain the ability to reorder for performance and parallelization. That is, let the set be an unordered set. I think the solution may be to provide an `UnorderedSet`, or `Sequence` class. This class would simply create a mapping from "insertion order" to "actual order" if it does decide to reorder. This is simply a convenience for programmers, so that they don't need to create this mapping themselves. Would that work for you?